ruuvi_examples/ruuvi_firmware/main.c

```diff
@@ -282,97 +296,149 @@ ret_code_t lis2dh12_int2_handler(const ruuvi_standard_message_t message)
+    nrf_delay_ms(1000u);  }
+  
+
+  // start watchdog now incase some initalization hangs up
+//  init_watchdog(NULL);  // watchdog_default_handler ONLY LOG_ERROR and returns to ??? 
+               // do something exciting , wait a few seconds and then keep going (i.e. never permit RESET!)       ToDo (_)
 
-  // Initialize log.
-  err_code |= init_log();
+//for ( int16_t i=0; i<13; i++){ RED_LED_ON; nrf_delay_ms(100u); RED_LED_OFF; nrf_delay_ms(200u); } // we got this far DEBUGING
 
-  // Setup leds. LEDs are active low, so setting high them turns leds off.
-  err_code |= init_leds();      // INIT leds first and turn RED on.
-  nrf_gpio_pin_clear(LED_RED);  // If INIT fails at later stage, RED will stay lit.
+  if( init_log() )                                                              init_status |=LOG_FAILED_INIT;       
+   else NRF_LOG_DEBUG("LOG initalized \r\n"); // subsequent initalizations assume log is working
 
-  //Init NFC ASAP in case we're waking from deep sleep via NFC (todo)
-  set_nfc_callback(app_nfc_callback);
-  err_code |= init_nfc();
 
-  if (NRF_SUCCESS == init_sensors()) { model_plus = true; }
+  battery_voltage_init();  // function with no return code (NEEDS rewrite)
+   uint16_t vbat= getBattery();
+   if(      vbat < BATTERY_MIN_V )                                             init_status |=BATTERY_FAILED_INIT;  
+    else NRF_LOG_DEBUG("BATTERY initalized \r\n"); 
 
-  // Initialize BLE Stack. Required in all applications for timer operation.
-  err_code |= init_ble();
+ if(init_sensors() == NRF_SUCCESS )  {   model_plus = true; // really need to be one at a time!!
+    NRF_LOG_DEBUG("Sensors initalized \r\n");  }
+
+  set_nfc_callback(app_nfc_callback);   // can this fail?
+//Init NFC ASAP in case we're waking from deep sleep via NFC (todo)
+ if( init_nfc() )                                                               init_status |= NFC_FAILED_INIT ; 
+  else NRF_LOG_DEBUG("NFC init \r\n");
+
+// Start (?) interrupts.  can this fail?
+ if( pin_interrupt_init() )                                                     init_status |= PIN_INT_FAILED_INIT ; 
+
+ if( pin_interrupt_enable(BSP_BUTTON_0, NRF_GPIOTE_POLARITY_HITOLO, button_press_handler)  )
+                                                                                init_status |= BUTTON_FAILED_INIT;
+// Initialize BLE Stack. Required for timer operation.
+ if(  init_ble() )                                                              init_status |= BLE_FAILED_INIT;  
   bluetooth_configure_advertisement_type(APPLICATION_ADVERTISEMENT_TYPE);
   bluetooth_tx_power_set(BLE_TX_POWER);
   bluetooth_configure_advertising_interval(ADVERTISING_INTERVAL_STARTUP);
 
-  // Initialize the application timer module.
-  err_code |= init_timer(main_timer_id, MAIN_LOOP_INTERVAL_RAW, main_timer_handler);
-
-  // Initialize RTC.
-  err_code |= init_rtc();
+  if( init_timer(main_timer_id, MAIN_LOOP_INTERVAL_RAW, main_timer_handler) )   init_status |= TIMER_FAILED_INIT;

```

Comment:
A matter of preference:
I prefer enclosing anything after if-clause In curly braces, i.e. 
`if( init_timer(main_timer_id, MAIN_LOOP_INTERVAL_RAW, main_timer_handler) )  { init_status |= TIMER_FAILED_INIT; }`.
This helps against some scoping errors.

Reply:
Unnecessary noise,  without braces indicates a very simple action is the result of the if.
but if you insist?

Reply:
Scrin: which line?

Reply:
The line this comment is attached to, which is as of now:
`if( init_timer(main_timer_id, MAIN_LOOP_INTERVAL_RAW, main_timer_handler) )   init_status |= TIMER_FAILED_INIT;`